src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -334,8 +350,35 @@ public int onErrorHandling() {
       return 1;
     }
 
+    /**
+     * Gets the custom template specified by user which can be either a templateId or a full custom template path
+     *
+     * @return String of templateId or custom template path
+     */
+    public String getCustomTemplate() {
+      return Strings.isNullOrEmpty(templateId) ? customTemplatePath :
+        String.format("projects/%s/inspectTemplates/%s", getProject(), templateId);
+    }
 
     public void validate(FailureCollector collector, Schema inputSchema) {
+      if (customTemplateEnabled) {
+        if (!containsMacro(CUSTOM_TEMPLATE_PATH_NAME) && Strings.isNullOrEmpty(customTemplatePath)) {
+          if (!containsMacro(TEMPLATE_ID_NAME) && Strings.isNullOrEmpty(templateId)) {
+            collector.addFailure("Must specify one of template Id or Custom Template Path", "")
+              .withConfigProperty(CUSTOM_TEMPLATE_ENABLED_NAME);
+          }
+        }
+        if (!containsMacro(TEMPLATE_ID_NAME) && Strings.isNullOrEmpty(templateId)) {
```

Comment:
Is this part of the code same as above check?

Reply:
This check is the opposite of the one above.

Reply:
This condition is the opposite of above. It checks whether both of custom template fields are filled. I simplified the conditions in the last updated PR.